.github/workflows/rust.yml

```diff
@@ -0,0 +1,35 @@
+name: Rust
+
+on:
+  push:
+    paths:
+      - "**.rs"
+      - "**.yml"
+      - "**.toml"
+      - "**.wit"
+  pull_request:
+    branches: [ "main" ]
+    paths:
+      - "**.rs"
+      - "**.yml"
+      - "**.toml"
+      - "**.wit"
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, windows-latest, macos-latest ]
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions-rs/toolchain@v1
```

Comment:
`actions-rs/toolchain` is no longer maintained. Use `https://github.com/dtolnay/rust-toolchain` instead.